NewHorizons/Components/NHGameOverManager.cs

```diff
@@ -134,5 +140,59 @@ private void LoadCreditsScene(GameOverModule gameOver)
                     break;
             }
         }
+
+        private void LoadCustomCreditsScene(GameOverModule gameOver, IModBehaviour mod)
+        {
+            var fromScene = LoadManager.GetCurrentScene();
+            var toScene = OWScene.Credits_Fast;
+
+            LoadManager.LoadScene(toScene, LoadManager.FadeType.ToBlack);
+
+            // We need to do this so we can unsubscribe from within the lambda.
+            LoadManager.SceneLoadEvent completeCreditsLoad = null;
+
+            completeCreditsLoad = (fromScene, toScene) =>
+            {
+                // Patch new music clip
+                var musicSource = Locator.FindObjectsOfType<OWAudioSource>().Where(x => x.name == "AudioSource").Single();
+                if (mod is not null)
+                {
+                    AudioUtilities.SetAudioClip(musicSource, gameOver.audio, mod);
+                }
+                else
+                {
+                    // We can't load in custom music if an IModBehaviour cannot be provided. This should only happen if called via TryHijackDeathSequence().
+                    NHLogger.LogWarning("Credits called using TryHijackDeathSequence(), custom credits audio cannot not be loaded.");
+                    return;
+                }
+                musicSource.loop = gameOver.audioLooping;
+                musicSource._maxSourceVolume = gameOver.audioVolume;
+
+                // Janky wait until credits are built
+                Task.Run( () =>
```

Comment:
I would prefer this to be done in a patch on CreditsScrollSection.Play() so we wont need to do the janky wait

Reply:
Although now that I think of it, you'd need some other janky stuff to refer back to a static variable defining what music to play. I'm mostly confused as to why this is required since we're already waiting for the credits scene to load. Maybe u just need a Delay.FireOnNextUpdate() (classic)